### PR TITLE
Proxy bypass

### PR DESCRIPTION
**Pull request by thedoginthewok**
See issue https://github.com/larshp/abapGit/issues/3219

### CODE REVIEW

File: src/http/zcl_abapgit_proxy_config.clas.abap [hunk 1]
```diff
@@ -81,4 +94,14 @@ CLASS ZCL_ABAPGIT_PROXY_CONFIG IMPLEMENTATION.
         cv_proxy_url = rv_proxy_url ).
 
   ENDMETHOD.
+
+
+  METHOD bypass_proxy.
+
+    IF iv_repo_url IN mo_settings->get_proxy_bypass( ).
```
Comment (flaiker) [inline src/http/zcl_abapgit_proxy_config.clas.abap]: I haven't looked into this in detail but won't and empty range table cause every URL to be included in the proxy bypass?

Comment (g-back) [reply src/http/zcl_abapgit_proxy_config.clas.abap]: I can confirm this, an empty range would cause this to always evaluate to `true`

Comment (thedoginthewok) [reply src/http/zcl_abapgit_proxy_config.clas.abap]: Yes, you're right. Rookie mistake...

I'll fix it.

File: src/http/zcl_abapgit_proxy_config.clas.abap [hunk 2]
```diff
@@ -98,7 +98,12 @@ CLASS zcl_abapgit_proxy_config IMPLEMENTATION.
 
   METHOD bypass_proxy.
 
-    IF iv_repo_url IN mo_settings->get_proxy_bypass( ).
+    DATA lt_proxy_bypass TYPE zif_abapgit_definitions=>ty_range_proxy_bypass_url.
+
+    lt_proxy_bypass = mo_settings->get_proxy_bypass( ).
+
+    IF lt_proxy_bypass IS NOT INITIAL
+    AND iv_repo_url IN mo_settings->get_proxy_bypass( ).
```
Comment (larshp) [inline src/http/zcl_abapgit_proxy_config.clas.abap]:
```suggestion
    AND iv_repo_url IN lt_proxy_bypass.
```

Comment (thedoginthewok) [reply src/http/zcl_abapgit_proxy_config.clas.abap]: That's what I meant to do, lol

Comment (thedoginthewok) [reply src/http/zcl_abapgit_proxy_config.clas.abap]: That's what I meant to do, lol

Comment (thedoginthewok) [reply src/http/zcl_abapgit_proxy_config.clas.abap]: Fixed!
Sorry about that.

**Comment by larshp** [comment]
welcome to abapGit as a contributor 🎉

**Pull request merged**